lib/utapi/utilities.js

```diff
@@ -49,7 +49,11 @@ function _listBucketMetrics(host,
             }
         });
     });
-    auth.client.generateV4Headers(request, '', accessKey, secretKey, 's3');
+    // TODO: cleanup with refactor of generateV4Headers
+    request.path = '/buckets';
+    auth.client.generateV4Headers(request, { Action: 'ListMetrics' },
+        accessKey, secretKey, 's3');
+    request.path = '/buckets?Action=ListMetrics';
```

Comment:
sorry it is not obvious to me : why do you change the request path after setting it before generating the headers?